src/scaife-widgets/urn.js

```diff
@@ -0,0 +1,115 @@
+export default class URN {
+  constructor(urn) {
+    this.absolute = urn;
+    this.delimiter = ':';
+    this.inclusiveRangeRe = /\d+-\d+$/;
+    this.subRangeRe = /\d+\.\d+-\d+\.\d+$/;
+    this.scheme = null;
+    this.nid = null;
+    this.nss = null;
+    this.work = null;
+    this.reference = null;
+    this.version = null;
+    this.hierarchy = null;
+    this.node = false;
+    this.range = false;
+    this.destructureUrn();
+    this.destructureReference();
+  }
+
+  static range(start, stop) {
+    return [...Array(stop - start + 1)].map((x, y) => y + start);
+  }
+
+  isNode() {
+    return this.node !== false;
+  }
+
+  isRange() {
+    return this.range !== false;
+  }
+
+  destructureUrn() {
+    const split = this.absolute.split(this.delimiter);
+    [this.scheme, this.nid, this.nss, this.work, this.reference] = split;
+    // TODO: Should have the trailing ':', see elsewhere.
```

Comment:
let's chat about this on the call today, and I solemnly swear I will circle back and add anything we discuss back on the PR 😄